doc/languages-frameworks/javascript.section.md

```diff
@@ -286,6 +298,28 @@ buildNpmPackage {
   npmConfigHook = importNpmLock.npmConfigHook;
 }
 ```
+:::
+
+:::{.example}
+##### `pkgs.importNpmLock` usage example with `fetcherOpts` {#javascript-buildNpmPackage-example-fetcherOpts}
+
+`importNpmLock` uses the following fetchers:
+
+- `pkgs.fetchurl` for `http(s)` dependencies
+- `builtins.fetchGit` for `git` dependencies
+
+It is possible to provide additional arguments as needed:
+
+```nix
+  # ... omited for clarity
```

Comment:
No, please make a self-contained example. We must expect those will be viewed completely out of context, and most often people don't even read the prose but just copy things around.

Reply:
Sphinx has [line highlighting](https://mystmd.org/guide/code#numbering-and-highlighting), maybe we should add something like this to our system eventually.